xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/user/api/XWikiGroupService.java

```diff
@@ -44,6 +44,17 @@
     void flushCache();
 
     /**
+     * Check whether the configuration specifies that every user is implicitly in XWikiAllGroup. Configured by the
+     * {@code xwiki.authentication.group.allgroupimplicit} parameter in {@code xwiki.cfg}.
+     *
+     * @param context the current XWiki context
+     * @return {@code true} if the group is implicit and all users should be by default in it, {@code false} if the
+     *         group behaves as all other groups, containing only the users/subgroups that are explicitly listed inside
+     *         the document.
+     */
+    public boolean isAllGroupImplicit(XWikiContext context);
```

Comment:
I would do the opposite actually: move isAllGroupImplicit to XWiki as a public method and call it in XWikiGroupServiceImpl#isAllGroupImplicit and XWiki#setUserDefaultGroup. Otherwise you are breaking XWikiGroupService interface and have to provide a default implementation for XWikiGroupService#isAllGroupImplicit method (with "default" keyword).